plugins/alerting/pkg/alerting/drivers/alerting_driver_dynamic.go

```diff
@@ -194,9 +200,125 @@ func (a *AlertingManager) Update(ctx context.Context, conf *alertops.AlertingCon
+			Namespace: a.gatewayRef.Namespace,
+		},
+	}
+	var numWorkerReplicas *int32
+	zero := int32(0)
+	err = a.k8sClient.Get(ctx, client.ObjectKeyFromObject(workerSvcData), workerSvcData)
+	if k8serrors.IsNotFound(err) {
+		numWorkerReplicas = &zero
+	} else if numWorkerReplicas == nil {
+		numWorkerReplicas = &zero
+	} else {
+		numWorkerReplicas = workerSvcData.Spec.Replicas
+	}
+	var wg sync.WaitGroup
+	annotationMutator := func(object client.Object) {
+		ann := object.GetAnnotations()
+		ann[opniReloadAnnotation] = fmt.Sprintf("%d", time.Now().UnixNano())
+		object.SetAnnotations(ann)
+	}
+	lg.Debugf("number of controller replicas : %d", *numReplicas)
+	lg.Debugf("number of worker replicas : %d", *numWorkerReplicas)
+	for i := 0; i < int(*numReplicas); i++ {
+		i := i // capture loop variable in closure
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			err = retry.OnError(retry.DefaultBackoff, k8serrors.IsConflict, func() error {
+				pod := &k8scorev1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      fmt.Sprintf("%s-%d", controllerSvcData.ObjectMeta.Name, i),
+						Namespace: a.gatewayRef.Namespace,
+					},
+				}
+				err := a.k8sClient.Get(ctx, client.ObjectKeyFromObject(pod), pod)
+				if err != nil {
+					lg.Error(err)
+					return err
+				}
+				clone := pod.DeepCopyObject().(client.Object)
+				annotationMutator(clone)
 
-	// FIXME? consider checking config map is updated before returning
+				cmp, err := patch.DefaultPatchMaker.Calculate(pod, clone,
+					patch.IgnoreStatusFields(),
+					patch.IgnoreVolumeClaimTemplateTypeMetaAndStatus(),
+					patch.IgnorePDBSelector(),
+				)
+				if err == nil {
+					if cmp.IsEmpty() {
+						return status.Error(codes.FailedPrecondition, "no changes to apply")
+					}
+				}
+				return a.k8sClient.Update(ctx, clone)
+			})
+			if err != nil {
+				lg.Error(err)
+			}
+		}()
+	}
+	for j := 0; j < int(*numWorkerReplicas); j++ {
+		j := j // capture loop variable in closure
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			err = retry.OnError(retry.DefaultBackoff, k8serrors.IsConflict, func() error {
+				pod := &k8scorev1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      fmt.Sprintf("%s-%d", workerSvcData.ObjectMeta.Name, j),
+						Namespace: a.gatewayRef.Namespace,
+					},
+				}
+				err := a.k8sClient.Get(ctx, client.ObjectKeyFromObject(pod), pod)
+				if err != nil {
+					lg.Error(err)
+					return err
+				}
+				clone := pod.DeepCopyObject().(client.Object)
+				annotationMutator(clone)
 
+				cmp, err := patch.DefaultPatchMaker.Calculate(pod, clone,
+					patch.IgnoreStatusFields(),
+					patch.IgnoreVolumeClaimTemplateTypeMetaAndStatus(),
+					patch.IgnorePDBSelector(),
+				)
+				if err == nil {
+					if cmp.IsEmpty() {
+						return status.Error(codes.FailedPrecondition, "no changes to apply")
+					}
+				}
+				return a.k8sClient.Update(ctx, clone)
+			})
+			if err != nil {
+				lg.Error(err)
```

Comment:
Should probably do something with this error, maybe just have more descriptive logging if you want this to be best-effort and not return an error